### PR TITLE
[hilariously wip]rewrites character saving/loading, adds support for character-specific antagonism preferences. potentially adds json backing up of character/exporting to compatible codebases (read: currently none)

### PR DESCRIPTION
**Pull request by silicons**
oh yeah this isn't done i'll be closing and reopening this later on finish for a new pr with a less low-effort description

organized like rp server's character setup but obviously, not like that, because that's also a dumpster fire, even if it's prettier than the current dumpster fire (before this is done)

## why

**i'm adding every body marking, ear, tail, almost every species icon base and *lots more* from rp later because the players want funny characters and i will deliver i guess** so haha protogens/vox/other funny things soon(tm) but i want to get this done first because the code is going to turn from a dumpster fire into a rotting crater if i do that without this

this is needed for the human appearance datum code refactor later where i buff the shit out of the "change" part of "changeilng"

also rushing this off so proteans won't be as awful code

## tldr on features

character list is now to the left of the window in a list to not clutter the main window. the other option is a popup but i hate popups and i'm sure you all do too

character specific antagonist and fetish content preference overrides will be added, toggleable to be on per character

players will be able to json save/load characters. fully sanitized, don't try anything funny

migrations being handled by each collection also adds a list for error strings to be added to, for logging and feedback purposes (so the player knows when a migration failed)

## tldr on code

character slots are now a config

persistent client variables like admin muting is moved to its own datum which is less awful than being on preferences

most of the code will be commented when i'm done because preferences is a trainwreck

preferences_collections holds most data and saves/loads to var/list/preferences on /datum/preferneces and handles the saving/loading/whatnot because i'm allergic to the current 1500 line proc, which again, is a trainwreck
each collection is also responsible for sanitization and copying its data to character when a character is being created for roundstart/latejoin

adds support for json saving/loading of characters because porting my characters to rp is a pain in the ass

reworks how saving works, everything now uses savekeys. a migration will ensure (most) data is not lost through the transition. each collection has its own savekey.

structure is now 

`/metadata` for general metadata
`/character#/metadata` for character metadata
`/character#/savekey` for each collection of settings

this is pain

### CODE REVIEW

**Comment by DeltaFire15** [comment]
Added merge conflict label due to current merge conflicts.

**Comment by DeltaFire15** [comment]
Marking this as stale due to the last commit having been nearly a month ago (and it's also merge conflicting yet again)

**Comment by DeltaFire15** [comment]
Closing this PR due to staleness & CI-failing state. Feel free to reopen it once work resumes.
